programs/cardinal-namespaces/src/instructions/update_entry_mint_metadata.rs

```diff
@@ -9,12 +12,30 @@ pub struct UpdateEntryMintMetadataCtx<'info> {
     namespace: Box<Account<'info, Namespace>>,
     #[account(constraint = entry.namespace == namespace.key() @ ErrorCode::InvalidNamespace)]
     entry: Box<Account<'info, Entry>>,
+    #[account(constraint = approve_authority.key() == namespace.approve_authority.unwrap() @ ErrorCode::InvalidApproveAuthority)]
+    approve_authority: Signer<'info>,
     #[account(mut)]
     certificate_mint_metadata: UncheckedAccount<'info>,
     token_metadata_program: UncheckedAccount<'info>,
 }
+#[derive(AnchorSerialize, AnchorDeserialize, Clone, Default)]
+pub struct Creator {
+    pub address: Pubkey,
+    pub verified: bool,
+    // In percentages, NOT basis points ;) Watch out!
+    pub share: u8,
+}
+
+#[derive(AnchorSerialize, AnchorDeserialize, Clone, Default)]
+pub struct UpdateMetadataArgs {
```

Comment:
```suggestion
pub struct UpdateMetadataIx {
```
just convention we have in the other instruction handlers :)